Sources/SpeziBluetooth/CoreBluetooth/BluetoothPeripheral.swift

```diff
@@ -218,22 +221,30 @@ public actor BluetoothPeripheral: BluetoothActor { // swiftlint:disable:this typ
         peripheral.delegate = delegate
     }
 
-    /// Establish a connection to the peripheral.
+    /// Establish a connection to the peripheral and wait until it is connected.
     ///
     /// Make a connection to the peripheral.
     ///
-    /// - Note: This method returns as soon as the request to connect was processed locally. It does
-    ///     not wait till the connection was completed successfully.
-    ///
     /// - Note: You might want to verify via the ``AdvertisementData/isConnectable`` property that the device is connectable.
-    public func connect() {
-        guard let manager else {
-            logger.warning("Tried to connect an orphaned bluetooth peripheral!")
-            return
-        }
-
-        manager.assumeIsolated { manager in
-            manager.connect(peripheral: self)
+    public func connect() async throws {
+        try await connectAccess.waitCheckingCancellation()
+
+        try await withTaskCancellationHandler {
+            try await withCheckedThrowingContinuation { continuation in
+                assert(connectContinuation == nil, "connectContinuation was unexpectedly not nil")
+                connectContinuation = continuation
+                guard let manager = self.manager else {
+                    logger.warning("Tried to connect an orphaned bluetooth peripheral!")
+                    return
+                }
+                manager.assumeIsolated {
+                    $0.connect(peripheral: self)
+                }
```

Comment:
All the `assumeIsolation` methods in SpeziBluetooth are a bit tricky but you did it right.

SpeziBluetooth implements all its entities using Actors to have nice isolation guarantees from the Swift compiler. However, it overwrites the actor executors and shares the same executor across all actors which helps immensely with synchronization and making sure some operations across actors are called in a certain order. In general also a pattern you see with CoreBluetooth, where you would normally use a single serial executor to dispatch all Bluetooth related work. 
Because we are sharing the same executors between actors we can just assume isolation here (because the Swift compiler can't know that statically).